vm/src/builtins.rs

```diff
@@ -64,10 +64,13 @@ fn builtin_ascii(obj: PyObjectRef, vm: &VirtualMachine) -> PyResult<String> {
     let mut ascii = String::new();
     for c in repr.value.chars() {
         if c.is_ascii() {
-            ascii.push(c)
+            ascii.push(c);
+        } else if (c as i64) < 0x10000 {
```

Comment:
Consider converting only once to `i64`, like this:
```rust
if c.is_ascii() {
    ...
} else {
    let c = c as i64;
    let hex = if c < 0x10000 {
        format!("\\u{:04x}", c)
    } else {
        format!("\\U{:08x}", c)    
    };
    ascii.push_str(&hex);
}

Reply:
Oh, this is great! Thanks!